tortoise/opinionhash/opinion_hasher.go

```diff
@@ -0,0 +1,68 @@
+package opinionhash
+
+import (
+	"github.com/spacemeshos/go-scale"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/hash"
+)
+
+var abstainSentinel = []byte{0}
+
+// New returns new instance of the opinion hasher.
+func New() *OpinionHasher {
+	hasher := hash.New()
+	return &OpinionHasher{enc: scale.NewEncoder(hasher), h: hasher}
+}
+
+// OpinionHasher is a utility for computing opinion hash.
+type OpinionHasher struct {
+	enc *scale.Encoder
+	h   hash.Hash
+}
+
+// WritePrevious aggregated hash.
+func (h *OpinionHasher) WritePrevious(hash types.Hash32) {
```

Comment:
questions about the APIs. does the order of calls to add previous/abstain/support matter?
does write of previous/abstain/support vs previous/support/abstain yield the same hash? if not, i think the code should enforce it.

Reply:
yes, all of that matters. order of supported blocks also matters

Reply:
i am not sure that it is really needed, the code would be much more complicated

Reply:
the previous code is simpler (prev hash + hash of sorted block ids) tho. in any case, it's fine as long as you've considered it.